dynamic_array.py

```
@@ -0,0 +1,113 @@
+        return self._n
+    
+    def is_empty(self):
+        """Checks if the array is empty."""
+        return self._n == 0
+    
+    def clear(self):
+        """Clears the array by resetting all elements while preserving the current capacity."""
+        self._n = 0
+        self._array = self._make_array(self._capacity)
+    
+    def __iter__(self):
```

Ruff (0.15.18)

[warning] Missing return type annotation for special method `__iter__`

(ANN204)